Plugins/Public/base_plugin/FactoryModule.cpp

```diff
@@ -1,81 +1,119 @@
 #include "Main.h"
 
-const char* RECIPE_NAMES[] =
-{ "Unknown", "recipe_make_dockmodule",
-	"recipe_make_jumpdrive_ii", "recipe_make_jumpdrive_iii", "recipe_make_jumpdrive_iv",
-	"recipe_make_hypscanner1", "recipe_make_hypscanner2", "recipe_make_hypscanner3",
-	"recipe_cloak_small", "recipe_cloak_medium", "recipe_cloak_large", "recipe_cloak_transport", "recipe_cloak_disruptor_1", "recipe_cloak_disruptor_2", "recipe_cloak_disruptor_3", "recipe_jdmatrix_1", 0 };
-
-const wchar_t* FACTORY_NAMES[] =
-{ L"Unknown", L"Unknown", L"Unknown", L"Unknown", L"Unknown",
-	L"Docking Module Factory", L"Jumpdrive Factory",
-	L"Hyperspace Scanner Factory", L"Cloaking Device Factory", L"Unknown", L"Unknown", L"Cloak Disruptor Factory", 0 };
-
-FactoryModule::FactoryModule(PlayerBase *the_base)
+FactoryModule::FactoryModule(PlayerBase* the_base)
 	: Module(0), base(the_base)
 {
 	active_recipe.nickname = 0;
 }
 
 // Find the recipe for this building_type and start construction.
-FactoryModule::FactoryModule(PlayerBase *the_base, uint the_type)
-	: Module(the_type), base(the_base)
+FactoryModule::FactoryModule(PlayerBase* the_base, uint nickname)
+	: Module(Module::TYPE_FACTORY), factoryNickname(nickname), base(the_base)
 {
 	active_recipe.nickname = 0;
+	for (wstring& craftType : factoryNicknameToCraftTypeMap[factoryNickname])
+	{
+		base->availableCraftList.insert(craftType);
+	}
 }
 
 wstring FactoryModule::GetInfo(bool xml)
 {
 	wstring info;
```

Comment:
doesn't this need to begin with `<TEXT>` if in XML mode?

Reply:
All the necessary prefixes and suffixes are handled inside the SendBaseStatus method in ClientCommands.cpp